cmd/controller/api/handlers.go

```diff
@@ -404,7 +404,7 @@ func GetPVCMetrics(w http.ResponseWriter, r *http.Request) {
 
 	var jsonData query.JSONDataWrapper
 	if name, isName := queryParams[query.Name]; isName {
-		jsonData = query.RetrievePVCMetrics(name[0])
+		jsonData = query.RetrieveResourceMetrics(query.PVCCheck, query.PVCType, name[0])
 	} else {
 		logrus.Errorf("wrong type of query for PVC, no name is given")
 	}
```

Comment:
Same as above.
Can we abstract all these to a common function/method as well for instance:
GetResourceMetrics(query Query) {}
I also suggest trying to create a method maybe.

struct type Query {
resourceType String,
query Query,
}

func (q Query) GetResourceHierarchy() {}
func (q Query) GetResourceMetrics() {}

Reply:
Done